### PR TITLE
IA-2308 Add advanced search in form submissions

### PR DESCRIPTION
**Pull request by bmonjoie**
Allow to filter the list by creation date and modification date.

Related JIRA tickets : IA-2308

## Self proofreading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [X] New translations have been added or updated if new strings have been introduced in the frontend
- [X] Are there enough tests

## Changes

Adding 4 new query parameters to `/api/instances/`:
- modificationDateFrom
- modificationDateTo
- sentDateFrom
- sentDateTo

Adding a new "Advanced settings" in the UI to allow filtering on those new parameters.

## How to test

1. Have an instance of Iaso with different submissions.
1. Navigate to the submission page
1. Click to show the advance parameter
1. Put values and click on "Search".

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/app/translations/fr.json [hunk 1]
```diff
@@ -224,6 +224,10 @@
     "iaso.forms.rejected": "Rejeté",
     "iaso.forms.restore.tooltip": "Récupérer le formulaire",
     "iaso.forms.shape": "Forme",
+    "iaso.instances.modificationDateFrom": "Date de modification à partir de",
```
Comment (hakifran) [inline hat/assets/js/apps/Iaso/domains/app/translations/fr.json]: The file(JSON) should be sorted by alphanumeric

File: hat/assets/js/cypress/integration/11 - submissions/list.spec.js [hunk 2]
```diff
@@ -476,6 +476,56 @@ describe('Submissions', () => {
         });
     });
 
+    it.only('advanced settings should filter correctly', () => {
+        goToPage();
+        cy.get('#advanced-settings').click();
```
Comment (hakifran) [inline hat/assets/js/cypress/integration/11 - submissions/list.spec.js]: This test generates an error
Screenshot from 2023-07-15 19-22-56

**Review by hakifran — APPROVED** [review]

File: hat/assets/js/cypress/integration/11 - submissions/list.spec.js (hunk 2 above)
Comment (beygorghor) [reply hat/assets/js/cypress/integration/11 - submissions/list.spec.js]: also we should remove the `.only`

File: hat/assets/js/cypress/integration/11 - submissions/list.spec.js [hunk 3]
```diff
@@ -476,6 +476,56 @@ describe('Submissions', () => {
...
+                            body: listFixture,
+                        });
+                    },
+                );
+            })
+            .as('getSubmissionsSearch');
+        cy.get('[data-test="search-button"]').click();
+        cy.wait('@getSubmissionsSearch').then(() => {
+            cy.wrap(interceptFlag).should('eq', true);
+        });
```
Comment (beygorghor) [inline hat/assets/js/cypress/integration/11 - submissions/list.spec.js]: Please check the code of users to test directly the XHR query params

**Review by beygorghor — CHANGES_REQUESTED** [review]
Please fix cypress

File: hat/assets/js/cypress/integration/11 - submissions/list.spec.js (hunk 2 above)
Comment (bmonjoie) [reply hat/assets/js/cypress/integration/11 - submissions/list.spec.js]: @hakifran it works on my machine... But I'll change the `id` to use `data-test` instead.

**Review by beygorghor — CHANGES_REQUESTED** [review]
two last things please:

- we can probably put those filters on 4 columns
- there is 1 pixel  of difference between columns 1, 2 and 3,4. 🪰 
Screenshot 2023-07-18 at 15 11 18

**Review by beygorghor — APPROVED** [review]

**Review by mathvdh — APPROVED** [review]
Ok for me on the python side

**Comment by beygorghor** [comment]
@cypress

**Pull request merged**